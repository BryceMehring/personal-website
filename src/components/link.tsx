import { Link as GatsbyLink } from 'gatsby';

interface Props {
  href: string;
  children: any;
  [key: string]: any;
}

export const Link = (props: Props): JSX.Element => {
  const { href, children, ...other } = props;
  if (!href.startsWith('/')) {
    return (
      <a href={href} {...other}>
        {children}
      </a>
    );
  }
  return (
    <GatsbyLink to={href} {...other}>
      {children}
    </GatsbyLink>
  );
};
