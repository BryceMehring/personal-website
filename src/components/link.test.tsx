import { Link } from './link';
import renderer from 'react-test-renderer';
import { describe, it, vi, expect } from 'vitest';

describe('link', () => {
  it('matches snapshot', () => {
    const wrapper = renderer.create(
      <Link href="test" data-internal-link>
        example
      </Link>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('matches snapshot with /', () => {
    const wrapper = renderer.create(
      <Link href="/test" data-external-link>
        example 2
      </Link>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
