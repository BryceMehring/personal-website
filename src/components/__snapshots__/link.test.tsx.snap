// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`link > matches snapshot 1`] = `
<a
  data-internal-link={true}
  href="test"
>
  example
</a>
`;

exports[`link > matches snapshot with / 1`] = `
<a
  data-external-link={true}
  href="/test"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  example 2
</a>
`;
